{
    "url": "https://api.github.com/repos/spack/spack/issues/4154",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/4154/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/4154/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/4154/events",
    "html_url": "https://github.com/spack/spack/issues/4154",
    "id": 226793826,
    "node_id": "MDU6SXNzdWUyMjY3OTM4MjY=",
    "number": 4154,
    "title": "Spack install of py-flake8, shows backports.configparser error",
    "user": {
        "login": "luigi-calori",
        "id": 5165864,
        "node_id": "MDQ6VXNlcjUxNjU4NjQ=",
        "avatar_url": "https://avatars.githubusercontent.com/u/5165864?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/luigi-calori",
        "html_url": "https://github.com/luigi-calori",
        "followers_url": "https://api.github.com/users/luigi-calori/followers",
        "following_url": "https://api.github.com/users/luigi-calori/following{/other_user}",
        "gists_url": "https://api.github.com/users/luigi-calori/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/luigi-calori/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/luigi-calori/subscriptions",
        "organizations_url": "https://api.github.com/users/luigi-calori/orgs",
        "repos_url": "https://api.github.com/users/luigi-calori/repos",
        "events_url": "https://api.github.com/users/luigi-calori/events{/privacy}",
        "received_events_url": "https://api.github.com/users/luigi-calori/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 73908754,
            "node_id": "MDU6TGFiZWw3MzkwODc1NA==",
            "url": "https://api.github.com/repos/spack/spack/labels/bug",
            "name": "bug",
            "color": "fc2929",
            "default": true,
            "description": null
        },
        {
            "id": 446619758,
            "node_id": "MDU6TGFiZWw0NDY2MTk3NTg=",
            "url": "https://api.github.com/repos/spack/spack/labels/python",
            "name": "python",
            "color": "c2e0c6",
            "default": false,
            "description": null
        }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 6,
    "created_at": "2017-05-06T18:36:37Z",
    "updated_at": "2019-12-02T20:16:42Z",
    "closed_at": "2019-12-02T20:16:42Z",
    "author_association": "CONTRIBUTOR",
    "active_lock_reason": null,
    "body": "On my linux Mint, I have installed with spack environment modules, configured with autoloading and use them to load python extensions.\r\nIt happens that \r\nspack install py-flake8\r\nmodule load py-flake8\r\nflake8\r\n\r\nresult in error:\r\n------\r\n------\r\n  File \"/kubuntu/home/lcalori/spack/RCM_test/deploy/rcm08/install/linux-linuxmint18-x86_64/gcc-5.4.0/py-configparser-3.5.0-gyfsxqxnkenrlpaam7zqlk4zox7uq4yi/lib/python2.7/site-packages/configparser.py\", line 12, in <module>\r\n    from backports.configparser import (\r\nImportError: No module named backports.configparser\r\n\r\nGoogling around, I found:\r\nhttps://pypi.python.org/pypi/backports/\r\n\r\nIt seems that, the proper__init__.py file is lacking from\r\n\r\nlib/python2.7/site-packages/backports\r\n\r\nIt probably relates to setuptools deciding to not install as it is a namespace package,\r\nas shown by the line in spack-build.out:\r\n\r\nSkipping installation of /kubuntu/home/lcalori/spack/RCM_test/deploy/rcm08/install/linux-linuxmint18-x86_64/gcc-5.4.0/py-configparser-3.5.0-gyfsxqxnkenrlpaam7zqlk4zox7uq4yi/lib/python2.7/site-packages/backports/__init__.py (namespace package)\r\n\r\nit seems a simple patch to setup.py can fix\r\n\r\n",
    "performed_via_github_app": null
}